src/test/func/malloc/malloc.cc

```diff
@@ -0,0 +1,118 @@
+#include <cerrno>
+#include <malloc.h>
+#include <snmalloc.h>
+
+using namespace snmalloc;
+
+void check_result(size_t size, size_t align, void* p, int err, bool null)
+{
+  if (errno != err)
+    abort();
+
+  if (null)
+  {
+    if (p != nullptr)
+      abort();
+  }
+  else
+  {
+    if (malloc_usable_size(p) < size)
```

Comment:
This is a non-standard function, so we can't expect it to be in any headers.  It isn't on FreeBSD or macOS.